src/datasources.js

```diff
@@ -1,3 +1,49 @@
+/**
+ * Asynchronously loads data sources from vellum.opts().core.dataSources
+ * Currently only supports fixtures
+ *
+ * Format in opts:
+ * dataSources: [
+ *     {
+ *          key: string : data source name (fixture, case, products...)
+ *          endpoint: function or string (URL)
+ *     }
+ * ]
+ *
+ * The endpoint function (or URL) should return the following format:
+ * [
+ *      {
+ *          sourceUri: string (used in the instance definition)
+ *          defaultId: string (used in instance definition)
+ *          intialQuery: string (used in nodeset)
+ *          name: string (text in the dropdown)
+ *          structure: nested dictionary of elements and attributes
+ *          {
+ *              element: {
+ *                  structure: {
+ *                      inner-element: {
+ *                          no_option: boolean (does not show up in the dropdown)
```

Comment:
Are you keeping this line? I think no need to (`no_option` can be an undocumented feature until we remove it). However, if you are then I suggest changing it to
```
no_option: true (does not show up in the dropdown)
```
since only a value of `true` will cause it to not show up in the dropdown.